rules/do-you-optimize-tinacms-project-for-clarity-performance-and-reliable-builds/rule.md

```diff
@@ -0,0 +1,109 @@
+
+Let’s explore how to structure your project effectively and apply best practices to boost performance both in runtime and during the build process.
+
+## 1. Structuring Your TinaCMS Architecture
+
+When working with large datasets or generating multiple subcomponents, following best practices is crucial to maintain performance and clarity.
+
+### Bad Practices:
+
+* **Making Individual Requests with Tina Client for Each Subcomponent:**
+
+  * This method can overwhelm the build process, particularly when dealing with a large number of subcomponents.
+* **Using Deeply Nested Schemas with Nested References:**
+
+  * Complex and deeply nested schemas increase the complexity of the project, making it harder to manage and more prone to build failures.
+  * They can also lead to inefficient data fetching, further slowing down both runtime and build processes.
+
+### Good Practices:
+
+* **Making a Single Request at a Top-Level Server Component and Passing Props Down:**
+
+  * Data fetched at the top level is cached by default, which enhances performance by reducing redundant API calls.
+  * Subcomponents can directly access the necessary data, eliminating the need for repeated requests.
+
+![✅ Figure: Good example - Single request at the top-level server and passing props down](2024-08-28_16-21-56.png)
+
+* **Caching Data at a Top-Level and Accessing It When Necessary:**
+
+  * If passing props is not feasible (e.g., when a component depends on Next.js router information), you should make a general top-level request, cache the data, and then access it directly from the cache within the component.
+  * This approach ensures efficient data retrieval and reduces the server load at build time.
+
+## 2. Improving Runtime Performance
+
+Optimizing runtime performance is key to delivering a fast and responsive user experience.
+
+### Bad Practices:
+
+* **Using Client-Side Requests Instead of Cached Values from Server-Side Components:**
+
+  * This approach bypasses pre-fetched, cached data, increasing server load and slowing down the application.
+  * It negates the benefits of static site generation, where data should be ready and waiting.
+
+### Common Practices:
+
+* **Utilizing Next.js caching via fetch:** 
+
+  * Next.js’s fetch API is designed to cache by default during static generation and server-side rendering.
+
+### Good Practices:
+
+* **Utilizing Caching at the Highest Level of the Application:**
+
+  * By caching data at the top level, you ensure that it is readily available throughout your application, reducing redundant data fetching and improving overall performance.
+  * This approach allows for more efficient data management and quicker response times.
+* **Using Dedicated Caching Packages:**
+
+  * [**Redis**:](https://redis.io/fr/solutions/cas-dutilisation/cache/) An in-memory data structure store, Redis is fast and versatile, making it ideal for caching data in memory. 
```

Comment:
@PothieuG 

The bold here is inconsistent with the next line...

Reply:
✅ Done